Add migration guide for 1.20 (#10562)
kyma-project/kyma: docs/migration-guides/1.19-1.20.md

# Migrate from 1.19 to 1.20

## Fluent Bit

> Kyma 1.20 comes with a redesigned configuration for Fluent Bit. It is redesigned to be up to date with the upstream chart and also be consistent with the rest of the configuration values. The latest configurations can be found [in the chart](https://github.com/kyma-project/kyma/blob/master/resources/logging/charts/fluent-bit/values.yaml#L134). If you have been using overrides for Fluent Bit, compare them with the latest configurations and adapt if necessary. It also now supports many more configuration options for the plugins coming out of the box. For example, you can set the `alias` field for all the input and output plugins to be used as the parent name for the metric. You can read more about it [here](https://docs.fluentbit.io/manual/administration/monitoring#configuring-aliases).  

Reviewer: Although it's not "wrong" or anything, if possible, it's nice to make the links more descriptive and avoid links like `here`, `this`,  as mentioned in https://kyma-project.io/community/guidelines/content/#links-links 🙂 

```suggestion
Kyma 1.20 comes with a redesigned configuration for Fluent Bit. It is redesigned to be up to date with the upstream chart and also be consistent with the rest of the configuration values. The latest configurations can be found in the [chart](https://github.com/kyma-project/kyma/blob/master/resources/logging/charts/fluent-bit/values.yaml#L134). If you have been using overrides for Fluent Bit, compare them with the latest configurations and adapt if necessary. It also now supports many more configuration options for the plugins coming out of the box. For example, you can set the `alias` field for all the input and output plugins to be used as the parent name for the metric. You can read more about it in this [Fluent Bit manual](https://docs.fluentbit.io/manual/administration/monitoring#configuring-aliases).  
```

An example migration looks like the following:

```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: logging-overrides
  namespace: kyma-installer
  labels:
    installer: overrides
    component: logging
    kyma-project.io/installation: ""
data:
  fluent-bit.conf.Output.http.enabled: "true"
  fluent-bit.conf.Output.http.Match : "*"
  fluent-bit.backend.http.host: "<Fluentd-endpoint from service key json>"
  fluent-bit.backend.http.port: "443"
  fluent-bit.backend.http.http_user: "<Fluentd-username from service key json>"
  fluent-bit.backend.http.http_passwd: "<Fluentd-password from service key json>"
  fluent-bit.backend.http.tls: "true"
  fluent-bit.backend.http.tls_verify : "false"
```

> Old overrides

Reviewer: Unless it's formatted as a title, or at least has `:` afterwards I think we should use complete sentences when introducing a code block/diagram/etc. so it's clear what is being referred to

```suggestion
These are the old overrides:
```

```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: logging-overrides
  namespace: kyma-installer
  labels:
    installer: overrides
    component: logging
    kyma-project.io/installation: ""
data:
  fluent-bit.config.outputs.http.enabled: "true"
  fluent-bit.config.outputs.http.match : "*"
  fluent-bit.config.outputs.http.host: "<Fluentd-endpoint from service key json>"
  fluent-bit.config.outputs.http.port: "443"
  fluent-bit.config.outputs.http.httpUser: "<Fluentd-username from service key json>"
  fluent-bit.config.outputs.http.httpPasswd: "<Fluentd-password from service key json>"
  fluent-bit.config.outputs.http.tls.enabled: "true"
  fluent-bit.config.outputs.http.tls.verify : "false"
```

New overrides

> This configuration can also be passed using the new additional keyword. Additional keyword exists also for inputs, filters, and parsers.

Reviewer 2: Does that mean "An additional keyword exists..." or "Additional keywords exist..." ? 
If it's just one additional keyword, we need the indefinite article "an".

Author: There are new fields called `additional`. I tried to make it more understandable.

> New overrides
> 
> This configuration can also be passed using the new additional keyword. Additional keyword exists also for inputs, filters, and parsers.

Reviewer: I think this remark would work better in the note format as it is some "additional information" related to the code sample that users should be aware of

```suggestion
These are the new overrides: 

> **NOTE:** This configuration can also be passed using the new additional keyword. Additional keyword exists also for inputs, filters, and parsers.
```

```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: logging-overrides
  namespace: kyma-installer
  labels:
    installer: overrides
    component: logging
    kyma-project.io/installation: ""
data:
  fluent-bit.config.outputs.additional: |
      [OUTPUT]
          Name                        http
          Match                       *
          Host                        <Fluentd-endpoint from service key json>
          Port                        443
          HTTP_User                   <Fluentd-username from service key json>
          HTTP_Passwd                 <Fluentd-password from service key json>
          tls                         On
          tls.verify                  false
```

> Additionally, we provide a way to keep the sensitive information - such as credentials for an output plugin - to be deployed inside a Secret instead of ending up in a ConfigMap on the cluster. You can pass the secret values as key/value pairs [here](https://github.com/kyma-project/kyma/blob/master/resources/logging/charts/fluent-bit/values.yaml#L135) and they will be populated to the Fluent Bit Pods as environment variables mounted from a Secret. Then, you can use the environment variable in the Fluent Bit configuration. We strongly recommend moving all the sensitive information in your configuration to this new field as they are kept more secure in a Secret.

Reviewer: "Information" is singular, so the verb should be in singular as well (unless this refers to the environment valuables?)
```suggestion
Additionally, we provide a way to keep the sensitive information - such as credentials for an output plugin - to be deployed inside a Secret instead of ending up in a ConfigMap on the cluster. You can pass the secret values as key/value pairs [here](https://github.com/kyma-project/kyma/blob/master/resources/logging/charts/fluent-bit/values.yaml#L135) and they will be populated to the Fluent Bit Pods as environment variables mounted from a Secret. Then, you can use the environment variable in the Fluent Bit configuration. We strongly recommend moving all the sensitive information in your configuration to this new field as it is kept more secure in a Secret.
```

> We have also included many native Fluent Bit [parsers](https://github.com/kyma-project/kyma/blob/master/resources/logging/charts/fluent-bit/templates/kyma-additions/_parsers.tpl). You can use the parsers by specifying a [parser filter](https://docs.fluentbit.io/manual/pipeline/filters/parser) for your workload or by annotating the Pods as explained [here](https://docs.fluentbit.io/manual/pipeline/filters/kubernetes#suggest-a-parser). 

Reviewer: Same note as above regarding [links](https://kyma-project.io/community/guidelines/content/#links-links)

```suggestion
We have also included many native Fluent Bit [parsers](https://github.com/kyma-project/kyma/blob/master/resources/logging/charts/fluent-bit/templates/kyma-additions/_parsers.tpl). You can use the parsers by specifying a [parser filter](https://docs.fluentbit.io/manual/pipeline/filters/parser) for your workload or by annotating the Pods as explained in the [Fluent Bit documentation](https://docs.fluentbit.io/manual/pipeline/filters/kubernetes#suggest-a-parser). 
```
